autogpt_platform/frontend/src/components/CustomEdge.tsx

```diff
@@ -87,89 +94,80 @@ export function CustomEdge({
     [getTForDistance, length, visualizeBeads],
   );
 
+  beadsRef.current = beads;
   useEffect(() => {
-    if (data?.beadUp === 0 && data?.beadDown === 0) {
+    const beadUp: number = data?.beadUp ?? 0;
+    const beadDown: number = data?.beadDown ?? 0;
+
+    if (
+      beadUp === 0 &&
+      beadDown === 0 &&
+      (beads.created > 0 || beads.destroyed > 0)
+    ) {
       setBeads({ beads: [], created: 0, destroyed: 0 });
       return;
     }
 
-    const beadUp: number = data?.beadUp ?? 0;
-
     // Add beads
-    setBeads(({ beads, created, destroyed }) => {
-      const newBeads = [];
-      for (let i = 0; i < beadUp - created; i++) {
-        newBeads.push({ t: 0, targetT: 0, startTime: Date.now() });
-      }
-
-      const b = setTargetPositions([...beads, ...newBeads]);
-      return { beads: b, created: beadUp, destroyed };
-    });
-
-    // Remove beads if not animating
-    if (visualizeBeads !== "animate") {
```

Comment:
As far as I could see, `visualizeBeads` is hardcoded to `"animate"` so this was dead code.

Reply:
Actually we should not animate static link or the first load of the graph,

Reply:
Is that something I need to fix in this PR or a note to fix sometime in the future? Afaik always-on animation is the current behavior.